src/controllers/scripting/legacy/controllerscriptinterfacelegacy.cpp

```diff
@@ -894,6 +894,14 @@ bool ControllerScriptInterfaceLegacy::isScratching(int deck) {
     return getValue(group, "scratch2_enable") > 0;
 }
 
+bool ControllerScriptInterfaceLegacy::isSoftStarting(int deck) {
+    return m_softStartActive[deck];
+}
+
+bool ControllerScriptInterfaceLegacy::isBraking(int deck) {
+    return m_brakeActive[deck];
+}
+
```

Comment:
this is user callable API. Add a bounds check and report with `m_pScriptEngineLegacy->logOrThrowError(...)`.